rca/projects/models.py

```diff
@@ -336,4 +345,227 @@ def get_context(self, request, *args, **kwargs):
+                    end_year = page.end_date.strftime("%Y")
+                    year = year + " - " + end_year
+
+            projects_formatted.append(
+                {
+                    "title": page.title,
+                    "image": page.hero_image,
+                    "link": page.url,
+                    "school": page.get_related_school(),
+                    "year": year,
+                    "listing_summary": page.listing_summary,
+                    "meta_heading": page.listing_title,
+                }
+            )
+        return projects_formatted
+
+    def get_active_filters(self, request):
+        return {
+            "type": request.GET.getlist("type"),
+            "subject": request.GET.getlist("subject"),
+            "school_or_centre": request.GET.getlist("school_or_centre"),
+        }
+
+    def get_extra_query_params(self, request, active_filters):
+        extra_query_params = []
+        for filter_name in active_filters:
+            for filter_id in active_filters[filter_name]:
+                extra_query_params.append(urlencode({filter_name: filter_id}))
+        return extra_query_params
+
+    def get_projects_query(self):
+        return (
+            ProjectPage.objects.live()
+            .public()
+            .descendant_of(self, inclusive=True)
+            .select_related("hero_image")
+        )
+
+    def get_results(self, request, projects_query, active_filters):
+        # Request filters
+        research_types = active_filters["type"]
+        subjects = active_filters["subject"]
+        school_or_centre = active_filters["school_or_centre"]
+
+        if research_types:
+            projects_query = projects_query.filter(
+                research_types__research_type_id__in=research_types
+            )
+
+        if subjects:
+            projects_query = projects_query.filter(subjects__subject_id__in=subjects)
+
+        if school_or_centre:
+            projects_query = projects_query.filter(
+                models.Q(related_school_pages__page_id__in=school_or_centre)
+                | models.Q(related_research_pages__page_id__in=school_or_centre)
+            )
+        return self._format_results(projects_query.distinct())
+
+    def get_context(self, request, *args, **kwargs):
+        page = request.GET.get("page", 1)
+        context = super().get_context(request, *args, **kwargs)
+        active_filters = self.get_active_filters(request)
+
+        # Send all the query params through to the context so they can be added
+        # to the pager links, E.G type=1&type=2&subject=1...
+        context["extra_query_params"] = self.get_extra_query_params(
+            request, active_filters
+        )
+
+        # Unfiltered Projects query.
+        projects_query = self.get_projects_query()
+
+        context["filters"] = self.get_filters(active_filters, projects_query)
+        context["featured_project"] = self._format_results([self.featured_project])[0]
```

Comment:
I think this need wrapping in an if statement as if there are no featured projects selected it will break